pybamm/expression_tree/operations/evaluate_julia.py

```diff
@@ -929,17 +927,17 @@ def get_julia_mtk_model(model, geometry=None, tspan=None):
 
     # Replace variables in the julia strings that correspond to pybamm variables with
     # their julia equivalent
-    for var_id, julia_id in variable_id_to_print_name.items():
+    for var, julia_id in variable_to_print_name.items():
         # e.g. boundary_value_right(cache_123456789) gets replaced with u1(t, 1)
-        cache_var_id = id_to_julia_variable(var_id, "cache")
+        cache_var_id = id_to_julia_variable(var.id, "cache")
```

Comment:
I am a bit confused, why is the `.id` here now?

Reply:
we still need id here to generate names like `cache_12345`. I guess it could go into the `id_to_julia_variable` function

Reply:
Ah, I get it. Looks good then!